lbrynet/core/Wallet.py

```diff
@@ -967,63 +968,34 @@ def _load_wallet(self):
 
     def _check_large_wallet(self):
         if len(self.wallet.addresses(include_change=False)) > 1000:
-            log.warning("Your wallet is excessively large, please follow instructions here: \
-https://github.com/lbryio/lbry/issues/437 to reduce your wallet size")
+            log.warning(("Your wallet is excessively large, please follow instructions here: ",
+                         "https://github.com/lbryio/lbry/issues/437 to reduce your wallet size"))
 
     def _load_blockchain(self):
         blockchain_caught_d = defer.Deferred()
 
-        def check_caught_up():
-            local_height = self.network.get_catchup_progress()
+        def on_update_callback(event, *args):
+            # This callback is called by lbryum when something chain
+            # related has happened
+            local_height = self.network.get_local_height()
             remote_height = self.network.get_server_height()
+            updated_blocks_behind = self.network.get_blocks_behind()
+            log.info(
+                'Local Height: %s, remote height: %s, behind: %s',
+                local_height, remote_height, updated_blocks_behind)
 
-            if remote_height == 0:
-                return
-
-            height_diff = remote_height - local_height
-
-            if height_diff <= 5:
-                self.blocks_behind = 0
-                msg = ""
-                if self._caught_up_counter != 0:
-                    msg += "All caught up. "
-                msg += "Wallet loaded."
-                alert.info(msg)
-                self._catch_up_check.stop()
-                self._catch_up_check = None
-                blockchain_caught_d.callback(True)
+            self.blocks_behind = updated_blocks_behind
+            if local_height != remote_height:
                 return
 
-            if height_diff < self.blocks_behind:
-                # We're making progress in catching up
-                self._lag_counter = 0
-                self.is_lagging = False
-            else:
-                # No progress. Might be lagging
-                self._lag_counter += 1
-                if self._lag_counter >= 900:
-                    self.is_lagging = True
-
-            self.blocks_behind = height_diff
+            assert self.blocks_behind == 0
+            self.network.unregister_callback(on_update_callback)
+            log.info("Wallet Loaded")
+            reactor.callFromThread(blockchain_caught_d.callback, True)
```

Comment:
the `on_update_callback` is called by lbryum in the wallet network daemon thread.  `callFromThread` moves the  control back to the reactor thread.